routes/badges.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2018, The Linux Foundation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+const asyncMiddleware = require('../middleware/asyncMiddleware');
+const express = require('express');
+const router = express.Router();
+const utils = require('../lib/utils');
+const badgeCalculator = require('../business/badgeCalculator');
+
+router.get('/:type/:provider/:namespace/:name/:revision', asyncMiddleware(getComponentBadge));
+
+async function getComponentBadge(request, result) {
```

Comment:
For the most part we have been putting the route function directly inline in the `get` parameters. One or two places where we need to share and implementation between multiple routes the function gets factored out. Suggest here merging the named function up into the last `get` param. Mostly for code consistency.

Reply:
Keeping it a separate function makes for easier testing (not that I've added a test for this 😆 ). For testing purposes I'm not a big fan of longer, anonymous functions.  This reminds me that I should add a test for this.

Reply:
fair enough. I'm fine changing the way the others are done.